### PR TITLE
zephyr: Fix possible memory leak in img_mgmt_impl_write_image_data

### PR DESCRIPTION
**Pull request by de-nordic**
The commit fixes memory leak that would happen in case when heap
allocation is used, for obtaining buffer for flash_img_context
and an error occurs in code that uses the obtained buffer.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Comment by de-nordic** [comment]
Hello @mlaz, @nvlsianpu, @utzig, can you take a look at this PR?

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c [hunk 1]
```diff
@@ -337,59 +337,83 @@ img_mgmt_impl_read(int slot, unsigned int offset, void *dst,
...
+/*
+ * The alloc_ctx and free_ctx are specifically provided for
+ * the img_mgmt_impl_write_image_data to allocate/free single flash_img_context
+ * type buffer.
+ * When heap is enabled these functions will operate on heap; when  heap is not
+ * allocated the alloc_ctx just returns pointer to static, global life-time
+ * variable, and free_ctx does nothing.
+ * CONFIG_HEAP_MEM_POOL_SIZE is C preprocessor literal.
+ */
+static inline struct flash_img_context *alloc_ctx()
```
Comment (utzig) [inline cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:349]: I would add the `void` parameter here.

Comment (utzig) [reply cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:349]: Any particular reason for this and `free_ctx` being `inline`?

Comment (de-nordic) [reply cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:349]: Will add void.
Regarding inline: these functions have been crated only for the `img_mgmt_impl_write_image_data` and each is called once; bodies of these functions are close to nothing (the  `CONFIG_HEAP_MEM_POOL_SIZE` is literal and will "optimize" one of paths out), so I have figured out that making all the call effort is pointless, just inline it.

**Review by utzig — APPROVED** [review]
LGTM

**Pull request merged**